ui/pages/destinations/[id]/index.js

```diff
@@ -248,17 +240,50 @@ function GrantCell({ grantsList, grant, destination, onRemove, onUpdate }) {
           <Disclosure defaultOpen={destinationPrivileges === undefined}>
             {({ open }) => (
               <>
-                <Disclosure.Button className='w-full'>
-                  <span className='flex items-center text-xs font-medium text-gray-500'>
-                    <ChevronRightIcon
-                      className={`${
-                        open ? 'rotate-90 transform' : ''
-                      } mr-1 h-3 w-3 text-gray-500`}
-                    />
-                    {`Namespaces (${namespacesPrivilegeMap.size})`}
-                  </span>
-                </Disclosure.Button>
+                <div className='mb-2 flex items-center space-x-2'>
+                  <input
+                    type='checkbox'
+                    className='h-4 w-4 rounded border-gray-300 text-blue-600 hover:cursor-pointer focus:ring-blue-500'
+                    ref={checkbox}
+                    checked={checked}
+                    onChange={() => {
+                      setSelectedNamespaces(
+                        checked ? [] : [...namespacesPrivilegeMap.keys()]
+                      )
+                      setChecked(!checked)
+                    }}
+                  />
+                  <Disclosure.Button className='w-full'>
+                    <span className='flex items-center text-xs font-medium text-gray-500'>
+                      {`Namespaces (${namespacesPrivilegeMap.size})`}
+                      <ChevronRightIcon
+                        className={`${
+                          open ? 'rotate-90 transform' : ''
+                        } ml-1 h-3 w-3 text-gray-500`}
+                      />
+                    </span>
+                  </Disclosure.Button>
+                </div>
+
                 <Transition show={open}>
+                  <div className='flex items-center justify-end'>
+                    <button
+                      className='rounded-md px-4 py-2 text-2xs font-medium text-red-500 hover:bg-red-100 disabled:cursor-not-allowed disabled:bg-white disabled:opacity-30'
+                      type='button'
+                      onClick={() => {
+                        selectedNamespaces.map(namespace =>
+                          handleRemove(namespace)
+                        )
+                        setSelectedNamespaces([])
+                      }}
+                      disabled={selectedNamespaces.length === 0}
+                    >
+                      <div className='flex flex-row items-center py-0.5'>
+                        <XIcon className='mr-1 mt-px h-3.5 w-3.5' />
+                        Bulk remove access
```

Comment:
```suggestion
                        Remove selected
```

Just a suggestion, not required at all